Unsupported multi table sink api, rollback to sink template

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

I previously used version 2.3.3 of SeaTunnel, and now I've upgraded to version 2.3.12, scheduling it through DolphinScheduler. However, I'm encountering failures with an error related to Hazelcast serialization. I haven't made any changes to my configuration except for updating the userparameter to usernamein the source section, and I'm using the new connector packages. The old version executes successfully, but the new version still fails.

### SeaTunnel Version

2.3.12

### SeaTunnel Config

```conf
mysql-connector-java-8.0.21.jar
connector-doris-2.3.12.jar
connector-jdbc-2.3.12.jar
connector-selectdb-cloud-2.3.12.jar
```

### Running Command

```shell
2026-01-28 16:01:18,844 INFO  [o.a.s.c.s.u.ConfigBuilder     ] [main] - Parsed config file: 
	{
	    "env" : {
	        "execution.parallelism" : 2,
	        "job.mode" : "BATCH"
	    },
	    "source" : [
	        {
	            "url" : "jdbc:mysql://rm-uf6275fste3bk3ubp.mysql.rds.aliyuncs.com:3306/aws_prod?serverTimezone=GMT%2b8",
	            "driver" : "com.mysql.cj.jdbc.Driver",
	            "connection_check_timeout_sec" : 100,
	            "username" : "******",
	            "password" : "******",
	            "query" : "SELECT `ID` as id, `ORGID` as orgid, `BINDID` as bindid, `CREATEDATE` as createdate, `CREATEUSER` as createuser, `UPDATEDATE` as updatedate, `UPDATEUSER` as updateuser, `PROCESSDEFID` as processdefid, `ISEND` as isend, `BANK` as bank, `BANK_ACCOUNT` as bank_account, `ALIPAY_ACCOUNT` as alipay_account, `MEDIA_BUSINESSLINE` as media_businessline, `MEDIA_PRODUCT` as media_product, `RESOURCE_TRACK` as resource_track, `TRACK_DETAIL` as track_detail, `BELONG_SUBJECT_NAME` as belong_subject_name, `MEDIA_ACCOUNT` as media_account, `MEDIA_PASSWORD` as media_password, `STATUS` as status, `MEDIA_ACCOUNT_REAL` as media_account_real, `MEDIA_PASSWORD_REAL` as media_password_real, `BELONG_SUBJECT_ID` as belong_subject_id FROM `bo_eu_supplier_account`",
	            "partition_column" : "id",
	            "partition_num" : 2,
	            "plugin_name" : "Jdbc"
	        }
	    ],
	    "sink" : [
	        {
	            "load-url" : "selectdb-cn-9cw4lrs3v01.selectdbfe.rds.aliyuncs.com:8080",
	            "jdbc-url" : "selectdb-cn-9cw4lrs3v01.selectdbfe.rds.aliyuncs.com:9030",
	            "username" : "******",
	            "password" : "******",
	            "cluster-name" : "new_cluster",
	            "table.identifier" : "dw_precision_marketing.ods_zd_bo_eu_supplier_account",
	            "sink.enable-2pc" : "true",
	            "sink.enable-delete" : "true",
	            "selectdb.config" : {
	                "file.type" : "json"
	            },
	            "plugin_name" : "SelectDBCloud"
	        }
	    ]
	}
```

### Error Exception

```log
Caused by: java.io.NotSerializableException: org.apache.seatunnel.connectors.selectdb.config.SelectDBConfig
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

[1769588613461.log](https://github.com/user-attachments/files/24904351/1769588613461.log)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-selectdb-cloud/src/main/java/org/apache/seatunnel/connectors/selectdb/config/SelectDBConfig.java
+++ b/seatunnel-connectors-v2/connector-selectdb-cloud/src/main/java/org/apache/seatunnel/connectors/selectdb/config/SelectDBConfig.java
@@ -23,12 +23,15 @@ import lombok.Getter;
 import lombok.Setter;
 import lombok.ToString;
 
+import java.io.Serializable;
 import java.util.Properties;
 
 @Setter
 @Getter
 @ToString
-public class SelectDBConfig {
+public class SelectDBConfig implements Serializable {
+
+    private static final long serialVersionUID = 1L;
 
     private String loadUrl;
     private String jdbcUrl;

--- a/seatunnel-connectors-v2/connector-selectdb-cloud/src/main/java/org/apache/seatunnel/connectors/selectdb/config/SelectDBConfig.java
+++ b/seatunnel-connectors-v2/connector-selectdb-cloud/src/main/java/org/apache/seatunnel/connectors/selectdb/config/SelectDBConfig.java
@@ -46,7 +46,7 @@ public class SelectDBConfig implements Serializable {
     private Integer bufferSize;
     private Integer bufferCount;
     private Integer flushQueueSize;
-    private Properties StageLoadProps;
+    private Properties stageLoadProps;
 
     public static SelectDBConfig loadConfig(ReadonlyConfig pluginConfig) {
         SelectDBConfig selectdbConfig = new SelectDBConfig();

--- a/seatunnel-connectors-v2/connector-selectdb-cloud/src/test/java/org/apache/seatunnel/connectors/selectdb/serialize/SelectDBConfigSerializableTest.java
+++ b/seatunnel-connectors-v2/connector-selectdb-cloud/src/test/java/org/apache/seatunnel/connectors/selectdb/serialize/SelectDBConfigSerializableTest.java
@@ -1,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.selectdb.serialize;
+
+import org.apache.seatunnel.connectors.selectdb.config.SelectDBConfig;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Properties;
+
+public class SelectDBConfigSerializableTest {
+
+    @Test
+    void testSelectDBConfigSerializable() throws Exception {
+        SelectDBConfig config = new SelectDBConfig();
+        config.setLoadUrl("localhost:8080");
+        config.setJdbcUrl("localhost:9030");
+        config.setClusterName("cluster");
+        config.setUsername("user");
+        config.setPassword("pwd");
+        config.setTableIdentifier("db.table");
+        config.setEnableDelete(true);
+        config.setLabelPrefix("label");
+        config.setEnable2PC(true);
+        config.setMaxRetries(3);
+        config.setBufferSize(1024);
+        config.setBufferCount(2);
+        config.setFlushQueueSize(10);
+        Properties stageLoadProps = new Properties();
+        stageLoadProps.setProperty("file.type", "json");
+        config.setStageLoadProps(stageLoadProps);
+
+        byte[] serialized;
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                ObjectOutputStream objectOutputStream =
+                        new ObjectOutputStream(byteArrayOutputStream)) {
+            objectOutputStream.writeObject(config);
+            objectOutputStream.flush();
+            serialized = byteArrayOutputStream.toByteArray();
+        }
+
+        SelectDBConfig deserialized;
+        try (ObjectInputStream objectInputStream =
+                new ObjectInputStream(new ByteArrayInputStream(serialized))) {
+            deserialized = (SelectDBConfig) objectInputStream.readObject();
+        }
+
+        Assertions.assertEquals(config.getLoadUrl(), deserialized.getLoadUrl());
+        Assertions.assertEquals(config.getJdbcUrl(), deserialized.getJdbcUrl());
+        Assertions.assertEquals(config.getClusterName(), deserialized.getClusterName());
+        Assertions.assertEquals(config.getUsername(), deserialized.getUsername());
+        Assertions.assertEquals(config.getPassword(), deserialized.getPassword());
+        Assertions.assertEquals(config.getTableIdentifier(), deserialized.getTableIdentifier());
+        Assertions.assertEquals(config.getEnableDelete(), deserialized.getEnableDelete());
+        Assertions.assertEquals(config.getLabelPrefix(), deserialized.getLabelPrefix());
+        Assertions.assertEquals(config.isEnable2PC(), deserialized.isEnable2PC());
+        Assertions.assertEquals(config.getMaxRetries(), deserialized.getMaxRetries());
+        Assertions.assertEquals(config.getBufferSize(), deserialized.getBufferSize());
+        Assertions.assertEquals(config.getBufferCount(), deserialized.getBufferCount());
+        Assertions.assertEquals(config.getFlushQueueSize(), deserialized.getFlushQueueSize());
+        Assertions.assertEquals(
+                config.getStageLoadProps().getProperty("file.type"),
+                deserialized.getStageLoadProps().getProperty("file.type"));
+    }
+}

--- a/seatunnel-connectors-v2/connector-selectdb-cloud/src/test/java/org/apache/seatunnel/connectors/selectdb/serialize/SelectDBConfigSerializableTest.java
+++ b/seatunnel-connectors-v2/connector-selectdb-cloud/src/test/java/org/apache/seatunnel/connectors/selectdb/serialize/SelectDBConfigSerializableTest.java
@@ -50,20 +50,7 @@ public class SelectDBConfigSerializableTest {
         stageLoadProps.setProperty("file.type", "json");
         config.setStageLoadProps(stageLoadProps);
 
-        byte[] serialized;
-        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-                ObjectOutputStream objectOutputStream =
-                        new ObjectOutputStream(byteArrayOutputStream)) {
-            objectOutputStream.writeObject(config);
-            objectOutputStream.flush();
-            serialized = byteArrayOutputStream.toByteArray();
-        }
-
-        SelectDBConfig deserialized;
-        try (ObjectInputStream objectInputStream =
-                new ObjectInputStream(new ByteArrayInputStream(serialized))) {
-            deserialized = (SelectDBConfig) objectInputStream.readObject();
-        }
+        SelectDBConfig deserialized = roundTrip(config);
 
         Assertions.assertEquals(config.getLoadUrl(), deserialized.getLoadUrl());
         Assertions.assertEquals(config.getJdbcUrl(), deserialized.getJdbcUrl());
@@ -81,5 +68,52 @@ public class SelectDBConfigSerializableTest {
         Assertions.assertEquals(
                 config.getStageLoadProps().getProperty("file.type"),
                 deserialized.getStageLoadProps().getProperty("file.type"));
+    }
+
+    @Test
+    void testSelectDBConfigSerializableWithNullStageLoadProps() throws Exception {
+        SelectDBConfig config = new SelectDBConfig();
+        config.setLoadUrl("localhost:8080");
+        config.setJdbcUrl("localhost:9030");
+        config.setUsername("user");
+        config.setPassword("pwd");
+        // stageLoadProps not set, keep it null
+
+        SelectDBConfig deserialized = roundTrip(config);
+
+        Assertions.assertEquals(config.getLoadUrl(), deserialized.getLoadUrl());
+        Assertions.assertEquals(config.getJdbcUrl(), deserialized.getJdbcUrl());
+        Assertions.assertEquals(config.getUsername(), deserialized.getUsername());
+        Assertions.assertEquals(config.getPassword(), deserialized.getPassword());
+        Assertions.assertNull(deserialized.getStageLoadProps());
+    }
+
+    @Test
+    void testSelectDBConfigSerializableWithEmptyStageLoadProps() throws Exception {
+        SelectDBConfig config = new SelectDBConfig();
+        config.setLoadUrl("localhost:8080");
+        config.setStageLoadProps(new Properties());
+
+        SelectDBConfig deserialized = roundTrip(config);
+
+        Assertions.assertEquals(config.getLoadUrl(), deserialized.getLoadUrl());
+        Assertions.assertNotNull(deserialized.getStageLoadProps());
+        Assertions.assertTrue(deserialized.getStageLoadProps().isEmpty());
+    }
+
+    private static SelectDBConfig roundTrip(SelectDBConfig config) throws Exception {
+        byte[] serialized;
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                ObjectOutputStream objectOutputStream =
+                        new ObjectOutputStream(byteArrayOutputStream)) {
+            objectOutputStream.writeObject(config);
+            objectOutputStream.flush();
+            serialized = byteArrayOutputStream.toByteArray();
+        }
+
+        try (ObjectInputStream objectInputStream =
+                new ObjectInputStream(new ByteArrayInputStream(serialized))) {
+            return (SelectDBConfig) objectInputStream.readObject();
+        }
     }
 }

